wrapper/wrapper

```diff
@@ -14,11 +14,23 @@ if [ -v MIR_SERVER_WAYLAND_HOST ]; then
   export MIR_SERVER_CONSOLE_PROVIDER
   host_wayland=$(dirname "${XDG_RUNTIME_DIR}")/${MIR_SERVER_WAYLAND_HOST}
   if [ ! -O "${host_wayland}" ] || [ ! -O "${host_wayland}.lock" ]; then
-    echo "ERROR: MIR_SERVER_WAYLAND_HOST [${MIR_SERVER_WAYLAND_HOST}] not set up"
+    echo "ERROR: MIR_SERVER_WAYLAND_HOST [${MIR_SERVER_WAYLAND_HOST}] set, but not available"
     exit 1
   fi
   ln -sf "${host_wayland}" "$XDG_RUNTIME_DIR"
   ln -sf "${host_wayland}.lock" "$XDG_RUNTIME_DIR"
+elif [ -v DISPLAY ] && ! snapctl is-connected x11; then
+  echo "ERROR: DISPLAY [${DISPLAY}] set, but you need to connect the x11 interface"
+  echo "Try: sudo snap connect $SNAP_NAME:x11"
+  exit 1
+elif [ $(id -u) != 0 ] && ! [ -v DISPLAY ] && ! snapctl is-connected login-session-control; then
+  echo "ERROR: You need one of:"
+  echo "  1. An X11 host specified by DISPLAY environment variable"
+  echo "  2. The login-session-control interface connected"
+  echo "     Try: sudo snap connect $SNAP_NAME:login-session-control"
```

Comment:
When I do that, it failed with "operation not permitted" and/or "Permission denied", even if I used `--console-provider=logind`. Something I'm missing? Anything more we should say here?

Reply:
So I should log out of my session / switch to GDM for it to work?